src/streams/receive.rs

```diff
@@ -381,8 +381,41 @@ impl StreamRx {
 
         let is_new_packet = register.update(seq_no, now, header.timestamp, clock_rate.get());
 
+        // Get the previous time for comparison
         let previous_time = self.last_time.map(|t| t.numer());
-        let time_u32 = extend_u32(previous_time, header.timestamp);
+
+        // Calculate the extended timestamp
+        let mut time_u32 = extend_u32(previous_time, header.timestamp);
+
+        // If we're coming out of a paused state and the timestamp would go backwards,
+        // incrementally adjust until we find a value that moves time forward
+        if self.paused && previous_time.is_some() && time_u32 < previous_time.unwrap() {
+            // In 32-bit RTP timestamps, adding 2^31 (MAX/2) flips to the other half of timestamp space
+            // This forces extend_u32 to produce a value in the next cycle
+            let half_cycle = 1u32 << 31;
+            let mut adjusted_ts = header.timestamp;
+
+            // Try at most twice (should be sufficient to find a suitable timestamp)
+            for _ in 0..2 {
+                // Add half a cycle to move to a different region of timestamp space
+                adjusted_ts = adjusted_ts.wrapping_add(half_cycle);
+
+                // Recalculate extended timestamp with adjusted value
+                let adjusted_time_u32 = extend_u32(previous_time, adjusted_ts);
+
+                // If this adjusted timestamp moves time forward, use it
+                if adjusted_time_u32 > previous_time.unwrap() {
+                    time_u32 = adjusted_time_u32;
+                    break;
+                }
+            }
+
+            // If we still couldn't find a timestamp that moves forward, force time ahead
+            if time_u32 < previous_time.unwrap() {
+                time_u32 = previous_time.unwrap() + 1;
```

Comment:
Won't this cause lots of weirdness. If we end up here, chances are we'll end up here again for the next frame and the resulting effect will be frames advancing extremely quickly e.g. instead of having 2700 ticks per frame for 90kHZ video we'll have 1 tick per frame until we can escape this condition

Reply:
Right. Let's just reset in this case.